### PR TITLE
solves issue #130

### PR DESCRIPTION
**Pull request by deepanshu44**
added next/prev links at the bottom of the page in test pages

### CODE REVIEW

**Comment by argyleink** [comment]
Lots of great work in there! 
I tried to match the comps a bit more, next pass i'll adjust their size on mobile but i need to stop reviewing for now. 

<details>

<summary>Desktop</summary>

Screen Shot 2022-01-12 at 4 11 06 PM

</details>

<details>

<summary>Mobile</summary>

Screen Shot 2022-01-12 at 4 11 16 PM

</details>

after the design review, i'll check out the rest more. i gave it a quick pass and didnt see any glaring issues. 
thanks for all your hackin on this!

**Comment by deepanshu44** [comment]
@argyleink Thanks for taking time to review this. 
Also, one thing, I forgot to remove backup files from my worktree while doing commits. 
I removed them and pushed a new commit in my [dev](https://github.com/deepanshu44/tooling.report/tree/dev) branch.
I also reordered your changes to the top in [another](https://github.com/deepanshu44/tooling.report/tree/refactor) branch.

File: static-build/index.tsx [hunk 1]
```diff
@@ -28,15 +28,38 @@ const toOutput: Output = {
...
+      shortDesc: testEntry[1].meta.shortDesc,
+    },
+  };
+}
+
 function addTestPages(tests: Tests, basePath = '') {
-  for (const [testPath, test] of Object.entries(tests)) {
+  const testEntries: [string, Test][] = Object.entries(tests);
+  const len = testEntries.length;
+  testEntries.forEach(([testPath, test], i) => {
```
Comment (argyleink) [inline static-build/index.tsx]: the for loop can provide an index as well no? we tried to keep the code style in for of loops, curious why you swapped it here?

File: static-build/index.tsx [hunk 2]
```diff
@@ -28,15 +28,38 @@ const toOutput: Output = {
...
     const testBasePath = basePath + testPath + '/';
+
+    const prevData: PaginationData =
+      i !== 0
+        ? getPrevOrNextData(testEntries[i - 1])
+        : getPrevOrNextData(testEntries[len - 1]);
+
+    const nextData: PaginationData =
```
Comment (argyleink) [inline static-build/index.tsx]: not sure how to handle it yet, but sometimes next and prev point to the same page and create duplicate links:

Screen Shot 2022-01-13 at 3 20 10 PM
 
2 ideas for a solution
1. have it be a link to the next/prev section collection landing page?
2. hide one of them depending if at the beginning or end of a collection
3. your idea here?

File: static-build/index.tsx [hunk 3]
```diff
@@ -28,15 +28,38 @@ const toOutput: Output = {
...
+      i !== len - 1
+        ? getPrevOrNextData(testEntries[i + 1])
+        : getPrevOrNextData(testEntries[0]);
+
     toOutput[testBasePath + 'index.html'] = renderPage(
-      <TestPage test={test} />,
+      <TestPage test={test} prev={prevData} next={nextData} />,
```
Comment (argyleink) [inline static-build/index.tsx]: i wonder if next and prev would be better as properties on test objects themselves? props wouldnt need to pass as far and each test would have it's own pointers? i dont have strong feelings here, just thinking out loud.

File: static-build/components/TestPagination/index.tsx [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+interface Props {
+  prev: PaginationData;
+  next: PaginationData;
+}
+
+const TestPagination: FunctionalComponent<Props> = ({ prev, next }: Props) => {
+  return (
+    <div class={$nav}>
+      <div>
+        {' '}
```
Comment (argyleink) [inline static-build/components/TestPagination/index.tsx]:
```suggestion
```

cruft? not sure the purpose of this

**Review by argyleink — COMMENTED** [review]
it overall looks good to me, pending a few of the questions and comments in here.

@jakearchibald you down to do a follow up review?

File: static-build/index.tsx (hunk 3 above)
Comment (deepanshu44) [reply static-build/index.tsx:58]: @argyleink Did you meant something like [this](https://github.com/deepanshu44/tooling.report/blob/refactor/missing-types.d.ts#L30)

File: static-build/index.tsx (hunk 2 above)
Comment (deepanshu44) [reply static-build/index.tsx]: What if we have both 1 and 2.
You can check [this](https://github.com/deepanshu44/tooling.report/tree/nav-test) where I developed a (sort of) prototype.

<details>
<summary> When at the beginning </summary>

goto-parent
</details>
<details>
<summary> When at the end </summary>

Screenshot_end



</details>

Comment (deepanshu44) [reply static-build/index.tsx]: @argyleink 
What if we have both 1 and 2.
You can check [this](https://github.com/deepanshu44/tooling.report/tree/nav-test) where I developed a (sort of) prototype.

<details>
<summary> When at the beginning </summary>

begining


</details>
<details>
<summary> When at the end </summary>

end
</details>

File: static-build/index.tsx (hunk 3 above)
Comment (deepanshu44) [reply static-build/index.tsx]: @argyleink Did you meant something like [this](https://github.com/deepanshu44/tooling.report/blob/dev/missing-types.d.ts#L30) 
See the two new entries added at the end of interface `Test`.

File: static-build/components/TestPagination/index.tsx (hunk 4 above)
Comment (deepanshu44) [reply static-build/components/TestPagination/index.tsx]: @argyleink I need to figure out where it came from

File: static-build/index.tsx (hunk 1 above)
Comment (deepanshu44) [reply static-build/index.tsx]: @argyleink I am really sorry, I should have followed your style.
I also really appreciate the time you give to my code, dude, Thanks.

File: static-build/index.tsx (hunk 3 above)
Comment (argyleink) [reply static-build/index.tsx]: yep like that!

File: static-build/index.tsx (hunk 2 above)
Comment (argyleink) [reply static-build/index.tsx]: nice, having both feels well rounded 🙂

**Review by argyleink — APPROVED** [review]
this is lookin super good! @jakearchibald did you wanna take a peek before this merges?

**Comment by jakearchibald** [comment]
@deepanshu44 this looks great to me! Can you remove the `package-lock.json` from this PR? Then I'll merge it.

**Comment by deepanshu44** [comment]
@jakearchibald @argyleink Thanks, but I encountered a bug while running the site on Firefox today. SVGs size seems to be broken. 
<details>
<summary> Broken size </summary>

Screenshot from 2022-02-09 00-05-13

</details>

I will fix it and send another push by tomorrow at most. It was my mistake to not realize that SVGs were already implemented in TestCrumbs component and all I had to do was just copy and paste. I sincerely apologize for my mistake.

**Comment by deepanshu44** [comment]
@argyleink @jakearchibald Pushed some changes. Everything seems to be working correctly in firefox and chrome.
<details>
<summary> Desktop </summary>

Screenshot from 2022-02-09 13-03-03
</details>
<details>
<summary> Mobile </summary>

Screenshot from 2022-02-09 13-15-15
</details>

**Comment by jakearchibald** [comment]
No need to apologize at all! We missed it in review too. I really appreciate all your work on this issue. Thanks! Shipping this now…

**Comment by jakearchibald** [comment]
Oh, one last thing. Can you remove the change to `package-lock.json` from this PR? I think we got crossed-wires earlier. When I said "remove it from this PR", I didn't mean that this PR should remove `package-lock.json` from the project, I meant that there shouldn't be a change to `package-lock.json` in this PR. This PR doesn't change `package.json` so there's no reason for it to make changes to `package-lock.json`.

**Comment by deepanshu44** [comment]
@jakearchibald Thanks :-) I restored the changes made by our Bot.
Edit: Latest commit  I made restores the `date modified` as well for package-lock.json.

**Comment by jakearchibald** [comment]
Perfect, thank you!

**Comment by deepanshu44** [comment]
Thank you, @argyleink, it's because of your active feedback this PR got accepted, and it's my first PR (apart from typos ones) :-)
Also thanks @jakearchibald 

I will be looking onto solving other active issues here as well in the future.

**Pull request merged**